R/plotCCA.R

```diff
@@ -0,0 +1,540 @@
+#'   (default: \code{ellipse_fill = TRUE})
+#'
+#' @param ellipse_alpha Number between 0 and 1 to adjust the opacity of ellipses.
+#'   (default: \code{ellipse_alpha = 0.2})
+#'
+#' @param ellipse_size Number specifying the size of ellipses.
+#'   (default: \code{ellipse_size = 0.1})
+#' 
+#' @param ellipse_linetype Discrete number specifying the style of ellipses
+#'   (default: \code{ellipse_linetype = 1})
+#' 
+#' @param vec_size Number specifying the size of vectors.
+#'   (default: \code{vec_size = 0.5})
+#' 
+#' @param vec_colour String specifying the colour of vectors.
+#'   (default: \code{vec_color = "black"})
+#' 
+#' @param vec_color Alias for `vec_colour`.
+#' 
+#' @param vec_linetype Discrete number specifying the style of vector lines.
+#'   (default: \code{vec_linetype = 1})
+#' 
+#' @param arrow_size Number specifying the size of arrows.
+#'   (defaults: \code{arrow_size = 0.25})
+#' 
+#' @param label_size Number specifying the size of text and labels.
+#'   (default: \code{label_size = 4})
+#' 
+#' @param label_colour String specifying the colour of text and labels.
+#'   (default: \code{label_color = "black"})
+#' 
+#' @param label_color Alias for `label_colour`.
+#' 
+#' @param vec_text TRUE or FALSE, should text instead of labels be used to label vectors.
+#'   (default: \code{vec_text = TRUE})
+#' 
+#' @param repel_labels TRUE or FALSE, should labels be repelled
+#'   (default: \code{repel_labels = TRUE})
+#' 
+#' @param ... additional parameters for plotting, inherited from
+#'   \code{\link[scater:plotReducedDim]{plotReducedDim}},
+#'   \code{\link[ggplot2:geom_label]{geom_label}} and
+#'   \code{\link[ggrepel:geom_label_repel]{geom_label_repel}}.
+#' 
+#' @details
+#' \code{plotRDA} and \code{plotCCA} create an RDA/CCA plot starting from the
+#' output of \code{\link[mia:runCCA]{CCA and RDA}} functions, two common methods
+#' for supervised ordination of microbiome data. Both
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-constructor]{TreeSummarizedExperiment}}
+#' and matrix objects are supported as input. When the input is a
+#' TreeSummarizedExperiment, this should contain the output of runRDA
+#' in the reducedDim slot and the argument \code{dimred} needs to be defined.
+#' When the input is an matrix, this should be returned as output from
+#' calculateRDA and the argument \code{dimred} needs not be defined. However,
+#' the first method is recommended because it provides the option to adjust
+#' aesthetics to the colData variables through the arguments inherited from
+#' \code{\link[scater:plotReducedDim]{plotReducedDim}}.
+#' 
+#' @return 
+#' A \code{ggplot2} object 
+#'
+#' @name plotCCA
+#'
+#' @examples
+#' # Load dataset
+#' library(miaViz)
+#' data("enterotype", package = "mia")
+#' tse <- enterotype
+#' 
+#' # Remove missing values from covariables
+#' tse <- tse[ , !is.na(tse$ClinicalStatus)]
+#' tse <- tse[ , !is.na(tse$Age)]
+#' tse <- tse[ , !is.na(tse$Gender)]
+#'  
```

Comment:
I think na.action parameter is more clear to use --> we can modify runRDA function and catch the `na.action` parameter. Without the parameter error occurs, which might be quite hard to solve. --> we could make a check and give informative messages (and maybe modify the default behavior of na.action)